### PR TITLE
fix(ecs): fix serviceArn token to have region/account in serviceArn

### LINKED ISSUE #18246: (ecs): Base Service tokenized Service Arn doesn't contain region or account.
### What is the problem?

The tokenized ARN](https://github.com/aws/aws-cdk/blob/ef8ab7245630f54b5a633925e27b34c31b712abc/packages/%40aws-cdk/aws-ecs/lib/base/base-service.ts#L429-L433) is using the clusterName and physicalName to generate the service ARN token.

1. The ARN should be using the serviceName vs the physicalName because the serviceName aligns with what the actual ARN would be.
2. The tokenized ARN should be fully qualified as a multiple parts so the ARN can be leveraged in `Ec2Service.fromEc2ServiceAttributes` or `FargateService.fromFargateServiceAttributes` to determine the region and account when deploying across regions.

### Reproduction Steps

This generates the token for the serviceArn.
```
const stack = new cdk.Stack();
const vpc = new ec2.Vpc(stack, 'MyVpc', {});
const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDef');

taskDefinition.addContainer('web', {
  image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
  memoryLimitMiB: 512,
});

const service = new ecs.Ec2Service(stack, 'Ec2Service', {
  cluster,
  taskDefinition,
});
```

`service.serviceArn` generates a token, if this is a different account or region the in the cause of using a code pipeline which is deploying across multiple accounts or regions leveraging `Ec2Service.fromEc2ServiceAttributes` with the service ARN would be used in the pipeline to get the service, as you directly pass the service across boundaries.

This means to get the serviceArn after the service definition you need to do something like below to get the serviceArn to not be a full token.  Also this is related to issue #16634.

```
const serviceArn = this.formatArn({
  service: 'ecs',
  resource: 'service',
  resourceName: `${cluster.clusterName}/${service.serviceName},
});
```

### What did you expect to happen?

I would expect the serviceArn look like this `arn:${Token[AWS.Partition.5]}:ecs:${Token[AWS.Region.6]}:${Token[AWS.AccountId.2]}:service/${Token[TOKEN.425]}/${Token[TOKEN.481]}`

Which will allow the region, account as well as the clusterName and serviceName to be populated correctly when deploying the token in.

### What actually happened?

The current serviceArn looks likes `${Token[TOKEN.489]}` which means using the `serviceArn` for `Ec2Service.fromEc2ServiceAttributes` doesn't set the region or account correctly.

### CDK CLI Version

N/A

### Framework Version

_No response_

### Node.js Version

v16.13.1

### OS

Mac

### Language

Typescript

### Language Version

_No response_

### Other information

This issue is seen in CDK tests so not related to specific CDK version.

### PR DESCRIPTION
**Pull request by tobytipton**
This fixes #18246 by changing the serviceArn to token each part vs creating one token which allows for using the serviceArn to get the environment.

Fixes #18246 
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-ecs/lib/base/from-service-attributes.ts [hunk 1]
```diff
@@ -47,7 +47,9 @@ export function fromServiceAtrributes(scope: Construct, id: string, attrs: Servi
     });
   } else {
     arn = attrs.serviceArn as string;
-    name = stack.splitArn(arn, ArnFormat.SLASH_RESOURCE_NAME).resourceName as string;
+    const resourceName = stack.splitArn(arn, ArnFormat.SLASH_RESOURCE_NAME).resourceName as string;
+    const resourceNameSplit = resourceName.split('/');
+    name = resourceNameSplit.length === 1 ? resourceName : resourceNameSplit[1];
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/lib/base/from-service-attributes.ts:52]: This part is overlapping with the changes that @jumic  is making in this PR https://github.com/aws/aws-cdk/pull/18140/files. 

Let's keep this PR to only changing `serviceArn` public property for Ec2 and Fargate Service that are created with the BaseService constructor. And we can leave the necessary changes for the `serviceArn` and `serviceName` public props for fromXXX() services to @jumic 's PR.

File: packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts [hunk 2]
```diff
@@ -3280,24 +3307,61 @@ describe('ec2 service', () => {
   });
 
   describe('When import an EC2 Service', () => {
-    test('with serviceArn', () => {
+    test('with old serviceArn', () => {
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts:3310]: Testing the new vs old service ARN format should be a part of https://github.com/aws/aws-cdk/pull/18140. It is out of scope for this one.

File: packages/@aws-cdk/aws-ecs/test/fargate/fargate-service.test.ts [hunk 3]
```diff
@@ -2111,24 +2111,64 @@ describe('fargate service', () => {
   });
 
   describe('When import a Fargate Service', () => {
-    test('with serviceArn', () => {
+    test('with old serviceArn', () => {
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/test/fargate/fargate-service.test.ts:2114]: These tests are also out of scope

File: packages/@aws-cdk/aws-ecs/lib/base/from-service-attributes.ts (hunk 1 above)
Comment (madeline-k) [reply packages/@aws-cdk/aws-ecs/lib/base/from-service-attributes.ts:52]: Same comment for the changes in `ec2-service.ts` and `fargate-service.ts`

File: packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts [hunk 4]
```diff
@@ -54,9 +54,36 @@ describe('ec2 service', () => {
...
+      });
+
+      expect(service.serviceArn).toContain(`service/${cluster.clusterName}/${service.serviceName}`);
+
+      const importService = ecs.Ec2Service.fromEc2ServiceAttributes(stack, 'importService', {
+        cluster: cluster,
+        serviceArn: service.serviceArn,
+      });
+      expect(importService.serviceArn).toContain(`service/${cluster.clusterName}/${service.serviceName}`);
+      expect(importService.serviceName).toEqual(service.serviceName);
```
Comment (madeline-k) [inline packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts:84]: this should be it's own test once the fromXXX() issues are fixed in https://github.com/aws/aws-cdk/pull/18140

**Review by madeline-k — DISMISSED** [review]
See my comment on https://github.com/aws/aws-cdk/pull/18140#issuecomment-1020700548 as well

File: packages/@aws-cdk/aws-ecs/lib/base/base-service.ts [hunk 5]
```diff
@@ -426,12 +426,12 @@ export abstract class BaseService extends Resource
       Annotations.of(this).addWarning('taskDefinition and launchType are blanked out when using external deployment controller.');
     }
 
-    this.serviceArn = this.getResourceArnAttribute(this.resource.ref, {
+    this.serviceName = this.getResourceNameAttribute(this.resource.attrName);
+    this.serviceArn = Stack.of(this).formatArn({
```
Comment (skinny85) [inline packages/@aws-cdk/aws-ecs/lib/base/base-service.ts:430]: Note: this *needs* to keep using the `getResourceArnAttribute()` helper - otherwise, things will break.

Comment (tobytipton) [reply packages/@aws-cdk/aws-ecs/lib/base/base-service.ts:430]: Is there a better approach?  The `getResourceArnAttribute()` generates a token which can't be parsed for region, account, clusterName or serviceName.  Which means when creating the new service to get the ARN which can be leveraged to be imported you have to do this to generate the code.

Comment (skinny85) [reply packages/@aws-cdk/aws-ecs/lib/base/base-service.ts:430]: The problem here is most likely the `${cluster.clusterName}` part we are referencing here. We are investigating right now to confirm that 🙂.

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
@tobytipton We'll need the build to be passing before we can provide another review of this. It looks like your update broke the tests in another package so please take a look at those. 

Additionally, please make sure that your PR title confirms to the conventional commit standard (fix, feat, chore) and that it is written in a style that will reflect correctly in the change log (See [Contributing Guide, Pull Requests](https://github.com/aws/aws-cdk/blob/v1-main/CONTRIBUTING.md#step-4-pull-request)). In a fix, the title should state the problem that is being fixed, now what the fix itself is.

File: packages/@aws-cdk/aws-ecs/lib/base/base-service.ts (hunk 5 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-ecs/lib/base/base-service.ts:470]: @madeline-k did you ever conclude your investigation about that suspicious `${cluster.clusterName}` expression causing problems?
